matrix_sdk/src/room/common.rs

```diff
@@ -0,0 +1,103 @@
+use matrix_sdk_common::api::r0::{
+    membership::{get_member_events, join_room_by_id, leave_room},
+    message::get_message_events,
+};
+use std::ops::Deref;
+
+use crate::{Client, Result, Room, RoomMember};
+
+/// A struct containing methodes that are common for Joined, Invited and Left Rooms
+#[derive(Debug, Clone)]
+pub struct Common {
+    inner: Room,
+    pub(crate) client: Client,
+}
+
+impl Deref for Common {
+    type Target = Room;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl Common {
+    /// Create a new `room::Common`
+    ///
+    /// # Arguments
+    /// * `client` - The client used to make requests.
+    ///
+    /// * `room` - The underlaying room.
+    pub fn new(client: Client, room: Room) -> Self {
+        // TODO: Make this private
+        Self {
+            inner: room,
+            client,
+        }
+    }
+
+    /// Leave this room.
+    ///
+    /// Only invited and joined rooms can be left
+    pub(crate) async fn leave(&self) -> Result<()> {
+        let request = leave_room::Request::new(self.inner.room_id());
+        let _response = self.client.send(request, None).await?;
+
+        Ok(())
+    }
+
+    /// Join this room.
+    ///
+    /// Only invited and left rooms can be joined via this method
+    pub(crate) async fn join(&self) -> Result<()> {
+        let request = join_room_by_id::Request::new(self.inner.room_id());
+        let _resposne = self.client.send(request, None).await?;
+
+        Ok(())
+    }
+
+    /// Sends a request to `/_matrix/client/r0/rooms/{room_id}/messages` and returns
+    /// a `get_message_events::Response` that contains a chunk of room and state events
+    /// (`AnyRoomEvent` and `AnyStateEvent`).
+    ///
+    /// # Arguments
+    ///
+    /// * `request` - The easiest way to create this request is using the
+    /// `get_message_events::Request` itself.
+    pub async fn messages(
```

Comment:
This one is still missing the [example](https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk/struct.Client.html#examples-4) it had.

Reply:
now it's there. :)

Reply:
Thanks, I'll let CI finish and merge if it doesn't complain.